src/Language/PureScript/AST/Exported.hs

```diff
@@ -144,3 +141,16 @@ isDctorExported ident (Just exps) ctor = test `any` exps
   test (TypeRef _ ident' Nothing) = ident == ident'
   test (TypeRef _ ident' (Just ctors)) = ident == ident' && ctor `elem` ctors
   test _ = False
+
```

Comment:
It's not duplicated code I'm worried about, it's about unnecessary API surface. The larger an API is, the harder it is for people to read the code, the more possibilities there are to consider when you're using it, and just in general the harder it is to work with. I do genuinely believe that if we added an option or an alternate variant to any function in the compiler like this because not all the work it was doing was necessary in every situation where it gets called, if we applied this principle consistently throughout the code base, we'd very quickly end up with a big mess.